Makefile

```diff
@@ -12,15 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+DOCKER ?= docker
+GOLANG_VERSION ?= 1.15.6
+BUILDIMAGE ?= nvidia/go-nvml-devel
+
 GEN_DIR = $(PWD)/gen
 PKG_DIR = $(PWD)/pkg
 GEN_BINDINGS_DIR = $(GEN_DIR)/nvml
 PKG_BINDINGS_DIR = $(PKG_DIR)/nvml
 
 SOURCES = $(shell find $(GEN_BINDINGS_DIR) -type f)
 
-.PHONY: all test clean
-.PHONY: bindings test-bindings clean-bindings
+.PHONY: $(TARGETS)
+TARGETS := $(NATIVE_TARGETS) $(DOCKER_TARGETS)
+NATIVE_TARGETS := all test clean bindings test-bindings clean-bindings
+DOCKER_TARGETS := $(patsubst %, docker-%, $(NATIVE_TARGETS))
```

Comment:
Does it work to define `.PHONY` before defining `TARGETS`? Also, does it work to define `TARGETS := $(NATIVE_TARGETS) $(DOCKER_TARGETS)` before `NATIVE_TARGETS` and `DOCKER_TARGETS`.  It could see how it would work if it was a `=` instead of a `:=`, but I thought `:=` actually evaluated as it was encountered (not lazily like normal `=`).

Reply:
You're right. Updating so that `.PHONY:` works as expected:

```
$ make -pq : | grep -E "^\.PHONY"
make: *** No rule to make target ':'.  Stop.
.PHONY: all test clean bindings test-bindings clean-bindings docker-all docker-test docker-clean docker-bindings docker-test-bindings docker-clean-bindings
```

Reply:
If `.PHONY` is set before `TARGETS` (even with immediate assignment), this does not work.